Improve ci with dotnet-releaser

xoofx opened the pull request:
Hey, 

this is a small PR to improve the usage of dotnet-releaser (thank you for using it! 🙂 )

* Adding build and test in Debug to dotnet-releaser (so that coverage takes it into account as well)
* Use `dotnet-releaser run` so that it builds on regular commits and publish only when there is a tag and it is the main branch

xoofx committed:
Improve usage of dotnet-releaser

phil-scott-78 commented:
Oh thank you!

I'm gonna have to sheepishly admit why it's configured like this - pure laziness. I'm wanting to deploy on each commit to main for now with using minver's autogenerated pre release version. 

Granted, I probably shouldn't be deploying five or six times in a night. But if I wanted to deploy on every push and tag for main is there a configuration to do so?

xoofx commented:
> I'm gonna have to sheepishly admit why it's configured like this - pure laziness. I'm wanting to deploy on each commit to main for now with using minver's autogenerated pre release version.
> Granted, I probably shouldn't be deploying five or six times in a night. But if I wanted to deploy on every push and tag for main is there a configuration to do so?

Oh right, ok! There is nothing currently to handle this case properly.

Just to understand more the use case, you would like to have a mode where on e.g every commit, it could push a preview NuGet package for example, but it would not create a GitHub release for them or do you expect actually to create a release?

phil-scott-78 commented:
You got it-  every commit to main is a pre-release that is pushed to nuget but no gh release is published. Then once I'm ready to bump it out of pre-release I tag the version and upon tag push that does the full release. 

PRs / pushes to branches other than main would get neither, of course. 

That model matches what we are doing with Spectre too, so it isn't just a model for my laziness :-)

xoofx commented:
> You got it- every commit to main is a pre-release that is pushed to nuget but no gh release is published. Then once I'm ready to bump it out of pre-release I tag the version and upon tag push that does the full release.

Ok understood, makes sense. I have updated dotnet-releaser which is adding `nuget.publish_draft` and I have updated this PR to leverage it, I haven't tested it but it should hopefully work as expected.

phil-scott-78 commented:
Right on, I appreciate that.

Happy to be a test case for this too.
